examples/seismic/tti/operators.py

```diff
@@ -110,9 +110,13 @@ def Gzz_centered_2d(model, field, costheta, sintheta, space_order):
     """
     order1 = space_order // 2
     Gz = -(sintheta * field.dx(fd_order=order1) +
-           costheta * field.dy(fd_order=order1))
-    Gzz = ((Gz * sintheta).dx(fd_order=order1).T +
-           (Gz * costheta).dy(fd_order=order1).T)
+              costheta * field.dy(fd_order=order1))
+    Gzz = (Gz * costheta).dy(fd_order=order1).T
+
+    # Add rotated derivative if angles are not zero. If angles are
+    # zeros then `0*Gz = 0` and doesn't have any `.dy` ....
```

Comment:
@mloubout actually, on second thought, the neat(er) way to go here would be subclassing SymPy's Zero and One and make them differentiable objects

Reply:
That would only work in some very special cases where we set the zero ourselves, in general, it's just gonna be random zero popping up from an expression we can't control so it'll still be a standard Sympy Zero if not just the zero number (or any number).

Reply:
@FabioLuporini thoughts?

Reply:
I'm not entirely sure there is no way of returning our own zero, but I agree this is work we don't have the time to undertake now, so for me it's a go provided that, as @georgebisbas pointed out, commit messages are rewritten to follow the naming conventions